core.ts

```diff
@@ -980,6 +988,56 @@ abstract class PinejsClientCoreTemplate<
 		return this.request(params, { method: 'DELETE' });
 	}
 
+	upsert(params: PinejsClientCoreFactory.UpsertParams) {
+		const { id, body, ...restParams } = params;
+
+		if (!isObject(id)) {
+			throw new Error('The id property must be an object');
+		}
+
+		const naturalKeyProps = Object.keys(params.id);
+		if (naturalKeyProps.length === 0) {
+			throw new Error(
+				'The id property must be an object with the natural key of the model',
+			);
+		}
+
+		if ('options' in params) {
+			throw new Error(`'options' are not supported in upserts`);
```

Comment:
Why not?

Reply:
I actually couldn't think of a use case and ended up adding a check for this to keep it simple.
I guess that the only extra handling that we would need will be the extra handling in case a $filter is already provided. Is that correct @Page- ?